### PR TITLE
chore(amplify_api): clean up some code to limit analysis infos

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
No user-facing change here, just some small changes in API category to reduce lint infos as we lost some pub points in last release for non-fatal static analysis violations.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_api/test/graphql_helpers_test.dart [hunk 1]
```diff
@@ -23,7 +23,7 @@ import 'package:flutter_test/flutter_test.dart';
 
 void main() {
   group('with ModelProvider', () {
-    final AmplifyAPI api = AmplifyAPI(modelProvider: ModelProvider.instance);
+    AmplifyAPI(modelProvider: ModelProvider.instance);
```
Comment (dnys1) [inline packages/amplify_api/test/graphql_helpers_test.dart:26]: Prefer to delete

File: packages/amplify_api/test/graphql_helpers_test.dart [hunk 2]
```diff
@@ -787,9 +787,9 @@ void main() {
 
   group('without ModelProvider', () {
     test('should handle no ModelProvider instance', () {
-      AmplifyAPI api = AmplifyAPI();
+      AmplifyAPI();
```
Comment (dnys1) [inline packages/amplify_api/test/graphql_helpers_test.dart:790]: Prefer to delete

File: packages/amplify_api/example/integration_test/graphql_tests.dart [hunk 3]
```diff
@@ -44,7 +44,7 @@ void main() {
       final req = GraphQLRequest<String>(
           document: graphQLDocument, variables: <String, String>{'id': id});
       final response = await Amplify.API.mutate(request: req).response;
-      Map data = jsonDecode(response.data!);
+      jsonDecode(response.data!);
```
Comment (dnys1) [inline packages/amplify_api/example/integration_test/graphql_tests.dart:47]: Prefer to delete or wrap with try catch so its intention is clear.

**Review by dnys1 — APPROVED** [review]

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_api/test/graphql_helpers_test.dart (hunk 1 above)
Comment (ragingsquirrel3) [reply packages/amplify_api/test/graphql_helpers_test.dart:26]: You want to delete this line completely? maybe I'm missing something, but this is needed to register the model provider in the plugin singleton, just the plugin as a value not needed after that (at least to test model helpers).

File: packages/amplify_api/test/graphql_helpers_test.dart (hunk 2 above)
Comment (ragingsquirrel3) [reply packages/amplify_api/test/graphql_helpers_test.dart:790]: needed to test failure logic of not using model provider. If deleted, uses previously registered one.

File: packages/amplify_api/example/integration_test/graphql_tests.dart (hunk 3 above)
Comment (ragingsquirrel3) [reply packages/amplify_api/example/integration_test/graphql_tests.dart]: done

**Pull request merged**